### PR TITLE
Stock-Journal: API, Summary, Done By

### LINKED ISSUE #827: Product ownership / roommate setup
While considering this for a roommate setup, some groceries would be shared but some are not. If my roommate and I both like to eat apples, it would be useful to track who is eating what. Perhaps there is a better way to denote shared vs. non-shared resources though.

### PR DESCRIPTION
**Pull request by fipwmaqzufheoxq92ebc**
Fixes #827 , #963 , #949 

I've implemented an API for the stock journal, available at `/api/stock/journal`.

For #827 and #949, I added the user to stock_log. The default for the migrations is 'admin'.

I also added a new Page (and API) "stock journal summary", which sums up the amounts grouped by product, user and transaction-type.

### CODE REVIEW

File: migrations/0115.sql [hunk 1]
```diff
@@ -0,0 +1,89 @@
+ALTER TABLE stock_log
+    RENAME TO stock_log_old;
+
+CREATE TABLE stock_log(
```
Comment (berrnd) [inline migrations/0115.sql:4]: Was this only about adding the new column `user_id`? Then just `ALTER TABLE stock_log ADD user_id INTEGER NOT NULL` would be enough...

**Review by berrnd — APPROVED** [review]
Thank you! :)

**Comment by berrnd** [comment]
Everything works, so I merged it - but what's the point of the new stock journal summary page?
The entries are grouped, ok - but without any date/time and/or undo possibility, I don't see any benefit of what this should say to me...
(For all the rest "analysis" things there is #64...)

**Comment by fipwmaqzufheoxq92ebc** [comment]
> Everything works, so I merged it - but what's the point of the new stock journal summary page?
> The entries are grouped, ok - but without any date/time and/or undo possibility, I don't see any benefit of what this should say to me...
> (For all the rest "analysis" things there is #64...)

For #827  "reports of 'consumed by'": My understanding is that he means a summary, e.g. how many apples he eats, and how many apples his roommate eats. It calculates the SUM(amount) per Product,User,Booking type. yes - that may be related to #64 .

**Comment by berrnd** [comment]
Ah, ok, thanks for the explanation - makes sense for this...

**Pull request merged**